src/dnaio/_core.pyx

```diff
@@ -150,6 +154,47 @@ cdef class SequenceRecord:
             )
         self._qualities = qualities
 
+    @property
+    def id(self):
+        cdef char *name
+        cdef size_t name_length
+        cdef size_t id_length
+        # Not yet cached is None
+        if self._id is None:
+            name = <char *>PyUnicode_DATA(self._name)
+            name_length = <size_t>PyUnicode_GET_LENGTH(self._name)
+            id_length = strcspn(name, "\t ")
+            if id_length == name_length:
+                self._id = self._name
+            else:
+                self._id = PyUnicode_New(id_length, 127)
+                memcpy(PyUnicode_DATA(self._id), name, id_length)
+        return self._id
+
+    @property
+    def comment(self):
+        cdef char *name
+        cdef size_t name_length
+        cdef size_t id_length
+        cdef char *comment_start
+        cdef size_t comment_length
+        # Not yet cached is None
+        if self._comment is None:
+            name = <char *>PyUnicode_DATA(self._name)
+            name_length = <size_t>PyUnicode_GET_LENGTH(self._name)
+            id_length = strcspn(name, "\t ")
+            if id_length == name_length:
+                self._comment = ""
+            else:
+                comment_length = name_length - (id_length + 1)
+                comment_start = name + id_length + 1
+                self._comment = PyUnicode_New(comment_length , 127)
+                memcpy(PyUnicode_DATA(self._comment), comment_start, comment_length)
+        # Cached but nothing is internally empty string, expose externally as None
```

Comment:
I don’t get the comment. Maybe just remove? Would be ok

Reply:
I will try to make it more clear in the comment. It is a rather odd design choice so it needs one, I think.